group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
+--- ActivityQuestion (//section/question)
+---- <no dedicated class, etree.Element is used> (//section/question/answer)
+--- ActivityAssessment (//section/assessment)
+---- <no dedicated class, etree.Element is used> (//section/assessment/answer)
+
+GroupActivity (paths relative to root element)
+- resources: [DottableDict(title, description, location)] - ./resources/document
+- grading_criteria: [DottableDict(title, description, location)] - ./resources/document[@grading_criteria="true"]     # attribute filter implicit
+- milestone_dates: {str: datetime.date} - ./dates/milestone
+- submissions: [DottableDict(id, title, description, location?)] - ./submissions/document
+- grade_questions: [ActivityQuestion] - //section/question
```

Comment:
@antoviaque if our assumptions about Activities ("Kickoff", "Strategy Table", "Ten Timeless Tests" in docs) being `GroupActivity` instances and taking into account there's only one `GroupActivity` per Group Project Xblock, than:

* resources - still belong to `GroupActivity` as in design mockups they are direct descendants of Activities
* grading_criteria - not sure we need it at all. It looks like it has something to do with group review; I'll check it out and update this post when/if I have more data
* milestone_dates - questionable. The idea here was to provide a *named* set of dates to be used in backend and shown in UI; designers can refer to these dates by including `<span class='milestone' data-date="date_name">` Since those dates are now in Project Navigator this might be not needed anymore.
* submissions - still belongs to `GroupActivity` as there're one set of uploads per Activity.
* grade_questions - it's just a cache of all questions, essentially it can be obtained by walking all sections, but for convenience it is built at initialization